starter-code/basic-algorithms.js

```diff
@@ -1,7 +1,48 @@
-// Names and Input
+let hacker1 = "Grégoire";
+console.log("The driver's name is " + hacker1);
 
+let hacker2 = "Clara";
+console.log("The navigator's name is " + hacker2);
 
-//Conditionals
+if (hacker2.length > hacker1.length) {
+  console.log(
+    "Yo, navigator got the longest name, it has" +
+      hacker1.length +
+      " characters"
+  );
+} else if (hacker1.length > hacker2.length) {
+  console.log(
+    "The Driver has the longest name, it has " + hacker1.length + " characters "
+  );
+} else if (hacker2.length === hacker1.length) {
+  console.log(
+    "wow, you both got equally long names, " + hacker1.length + " characters!! "
+  );
+}
 
+var hacker1Maj = hacker1.toUpperCase();
+var arrayofletter = [];
+for (var i = 0; i < hacker1Maj.length; i++) {
+  //console.log(hacker1Maj[i])
+  arrayofletter.push(hacker1Maj[i]);
+}
+console.log(arrayofletter.join(" "));
 
-// Lorem ipsum generator
+var reversed = arrayofletter.reverse();
+console.log(reversed);
+console.log(reversed.join(" "));
+
+var alpha = arrayofletter.sort();
+console.log(alpha);
+console.log(alpha.join(" "));
+
+a = "Grégoire".localeCompare("Clara");
+document.write(a + "<br>");
+
+if (a > 0) {
+  console.log("The driver's name goes first");
+} else if (a < 0) {
+  console.log("Yo, the navigator goes first definitely");
+} else if ((a = 0)) {
+  console.log("What?! You both got the same name?");
+}
```

Comment:
CORRECTION DE LA SUITE:
`var isItAPal = prompt("Enter a string")

checkPal = () => {

var test1 = isItAPal.toLowerCase().replace(/\s/g, '') // on retire les espaces avec .replace et une regex
var test2 = isItAPal.toLowerCase().split("").reverse().join("").replace(/\s/g, '') // on split a chaque lettre, on reverse, on join et on retire les espaces.
return test1.localeCompare(test2) // On utilise localCompare() qui va nous donner la valeur lexicographique des deux variables (test1 et test2). Si on a autre chose que 0 (localcompare renvoie -1, 0 ou 1) alors ce n'est pas un palindrome)
}

checkPal()

var loremIps = "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce porttitor ut tortor sit amet aliquam. Etiam odio risus, finibus ullamcorper orci quis, mollis consectetur magna. Pellentesque scelerisque turpis ac sapien euismod, sed vestibulum purus dapibus. Etiam vel dictum felis. Vivamus rutrum porta et nunc sit amet tincidunt. Proin molestie mollis et nulla vel rutrum. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Nullam vestibulum neque purus, non mollis purus et sagittis a. Fusce tempor efficitur tortor vitae porta. Sed at dui urna. Nam feugiat magna tortor, at blandit magna molestie id. Ut sit amet dui ac diam porttitor vulputate. Cras velit odio, feugiat congue varius nec, dignissim sed est. Mauris commodo accumsan justo a fringilla. Vestibulum eu urna at velit ullamcorper scelerisque. Aliquam dignissim risus nec eros malesuada, id feugiat leo egestas. Sed non imperdiet ipsum. Praesent eu erat ac purus mattis pellentesque. Integer ultricies imperdiet enim, a mattis dolor molestie eget. Nullam est lectus, consequat ut sem sit amet, dapibus commodo erat. Pellentesque porttitor metus a egestas ultricies. Praesent elementum mollis nulla a efficitur. Etiam pellentesque, felis quis hendrerit semper, urna nibh auctor dui, id egestas urna lacus vitae leo. Vivamus consectetur libero in nunc pretium, luctus bibendum elit tincidunt. Ut laoreet tellus sed risus imperdiet, eget condimentum mauris ultricies. Duis consectetur in elit vitae aliquet.Proin tristique, neque id hendrerit molestie, risus nibh pulvinar odio, nec accumsan est turpis vitae lacus. Aenean sapien tortor, hendrerit non lorem eu, laoreet accumsan leo. Praesent consectetur nisl et erat scelerisque ullamcorper. Pellentesque elementum mauris sit amet blandit faucibus. Nullam nunc augue, vulputate tempus tristique eu, dignissim at nisi. Interdum et malesuada fames ac ante ipsum primis in faucibus. Mauris luctus lacinia aliquam."

countingWords = () => {
var loremArr = loremIps.split(" ") // on split la string a chaque espace, ca nous renvoie un array et on peut facilement calculer la longueur ensuite.
console.log(loremArr.length)
var count = 0;
for (let i = 0; i < loremIps.split(" ").length; i++) {
if(loremArr[i] === "et") { // si un mot de du text correspond a "et" ajouter 1 a count.
count += 1
}
}
console.log("Et count is " + count)
}

countingWords()`